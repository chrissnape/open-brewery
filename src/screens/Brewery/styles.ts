import { StyleSheet } from 'react-native';
import { borderColour, padding, primaryColour } from '../../utils/styles';

export default StyleSheet.create({
  cardWrapper: {
    marginBottom: padding,
  },
  container: {
    padding,
  },
  showInfoButton: {
    borderBottomColor: borderColour,
    borderBottomWidth: 1,
    color: primaryColour,
    fontWeight: 'bold',
    marginBottom: padding,
    paddingVertical: padding,
    textAlign: 'center',
  },
});
