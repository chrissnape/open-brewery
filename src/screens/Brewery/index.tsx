import { StatusBar } from 'expo-status-bar';
import * as WebBrowser from 'expo-web-browser';
import React, { FC, Fragment, useState } from 'react';
import { Text, View, TouchableWithoutFeedback, ScrollView } from 'react-native';
import { Button, InfoRow } from '../../components/index';
import { Brewery } from '../../utils/types';
import styles from './styles';

type Props = {
  addToFavourites: (id: number) => void;
  favouriteBreweries: Array<number>;
  selectedBrewery: Brewery | null;
}

const BreweryScreen: FC <Props> = ({ addToFavourites, favouriteBreweries, selectedBrewery }): JSX.Element => {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  if (!selectedBrewery) {
    return <Text>Unable to find brewery</Text>
  }
  const { id, brewery_type, city, county_province, phone, postal_code, state,website_url } = selectedBrewery;
  const isFavourite = favouriteBreweries.includes(id);
  const showLabel  = isOpen ? 'Hide information' : 'Show more information';
  return (
    <Fragment>
      <StatusBar style="light" />
      <ScrollView>
        <View style={styles.container}>
          <View style={styles.infoRowWrapper}>
            <InfoRow
              keyLabel="City"
              label={city}
            />
          </View>
          <View style={styles.infoRowWrapper}>
            <InfoRow
              keyLabel="State"
              label={state}
            />
          </View>
          {county_province && (
            <View style={styles.infoRowWrapper}>
              <InfoRow
                keyLabel="County"
                label={county_province}
              />
            </View>
          )}
          <View style={styles.infoRowWrapper}>
            <InfoRow
              keyLabel="Postal Code"
              label={postal_code}
            />
          </View>
          {(phone || website_url) && (
            <Fragment>
              <TouchableWithoutFeedback onPress={() => setIsOpen(!isOpen)}>
                <Text style={styles.showInfoButton}>{showLabel}</Text>
              </TouchableWithoutFeedback>
              {isOpen && (
                <Fragment>
                  <View style={styles.infoRowWrapper}>
                    <InfoRow
                      keyLabel="Brewery Type"
                      label={brewery_type}
                    />
                  </View>
                  {phone && (
                    <View style={styles.infoRowWrapper}>
                      <InfoRow
                        keyLabel="Phone Number"
                        label={phone}
                      />
                    </View>
                  )}
                  {website_url && (
                    <View style={styles.infoRowWrapper}>
                      <InfoRow
                        keyLabel="Website URL"
                        label={website_url}
                        onPress={() => WebBrowser.openBrowserAsync(website_url)}
                      />
                    </View>
                  )}
                </Fragment>
              )}
            </Fragment>
          )}
        </View>
      </ScrollView>
      <View style={styles.buttonWrapper}>
        <Button label={isFavourite ? 'Remove from favourites' : 'Add to favourites'} onPress={() => addToFavourites(id)} primary={!isFavourite} />
      </View>
    </Fragment>
  );
}

export default BreweryScreen;
