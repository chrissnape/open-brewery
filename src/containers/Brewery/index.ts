import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import {
  addToFavourites as addToFavouritesAction,
} from '../../actions';
import { Brewery } from '../../screens';
import { AppState } from '../../store';

const mapStateToProps = (state: AppState) => {
  const { breweries } = state;
  return { ...breweries };
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
  addToFavourites: (id: number) => addToFavouritesAction(dispatch, id),
});

export default connect(mapStateToProps, mapDispatchToProps)(Brewery);
