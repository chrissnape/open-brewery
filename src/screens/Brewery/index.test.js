import React from 'react';
import renderer from 'react-test-renderer';
import Brewery from './index';

const props = {
  addToFavourites: () => {},
  favouriteBreweries: [],
  selectedBrewery: null,
}

describe('<Brewery />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Brewery { ...props } />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});