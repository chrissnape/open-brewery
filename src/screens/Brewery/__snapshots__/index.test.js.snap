// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Brewery /> renders correctly 1`] = `
<Text>
  Unable to find brewery
</Text>
`;
